--- /var/www/html/snipeit/app/Models/Location.php	2025-03-24 17:06:07.099202423 +0100
+++ /var/www/html/snipeit8/app/Models/Location.php	2025-03-26 16:14:28.456507220 +0100
@@ -8,7 +8,7 @@
 use App\Models\Traits\Searchable;
 use App\Models\User;
 use App\Presenters\Presentable;
-use DB;
+use Illuminate\Support\Facades\DB;
 use Illuminate\Database\Eloquent\Factories\HasFactory;
 use Illuminate\Database\Eloquent\Model;
 use Illuminate\Database\Eloquent\SoftDeletes;
@@ -29,11 +29,11 @@
         'address'       => 'max:191|nullable',
         'address2'      => 'max:191|nullable',
         'city'          => 'max:191|nullable',
-        'state'         => 'min:2|max:255|nullable',
+        'state'         => 'min:2|max:191|nullable',
         'country'       => 'min:2|max:191|nullable',
         'zip'           => 'max:10|nullable',
         'manager_id'    => 'exists:users,id|nullable',
-        'parent_id'     => 'non_circular:locations,id',
+        'parent_id'     => 'nullable|exists:locations,id|non_circular:locations,id',
     ];
 
     protected $casts = [
@@ -42,7 +42,7 @@
     ];
 
     /**
-     * Whether the model should inject it's identifier to the unique
+     * Whether the model should inject its identifier to the unique
      * validation rules before attempting validation. If this property
      * is not set in the model it will default to true.
      *
@@ -72,7 +72,7 @@
         'currency',
         'manager_id',
         'image',
-        'created_at',
+        'notes',
     ];
     protected $hidden = ['user_id'];
 
@@ -83,7 +83,7 @@
      *
      * @var array
      */
-    protected $searchableAttributes = ['name', 'address','address2', 'city', 'state', 'zip', 'created_at', 'ldap_ou', 'phone', 'fax','currency'];
+    protected $searchableAttributes = ['name', 'address', 'city', 'state', 'zip', 'created_at', 'ldap_ou', 'phone', 'fax', 'notes'];
 
     /**
      * The relations and their attributes that should be included when searching the model.
@@ -109,10 +109,11 @@
     {
 
         return Gate::allows('delete', $this)
-                && ($this->assets_count === 0)
-                && ($this->assigned_assets_count === 0)
-                && ($this->children_count === 0)
-                && ($this->users_count === 0);
+                && ($this->assets_count == 0)
+                && ($this->assigned_assets_count == 0)
+                && ($this->children_count == 0)
+                && ($this->accessories_count == 0)
+                && ($this->users_count == 0);
     }
 
     /**
@@ -253,6 +254,18 @@
         return $this->morphMany(\App\Models\Asset::class, 'assigned', 'assigned_type', 'assigned_to')->withTrashed();
     }
 
+    /**
+     * Establishes the accessory -> location assignment relationship
+     *
+     * @author A. Gianotto <snipe@snipe.net>
+     * @since [v3.0]
+     * @return \Illuminate\Database\Eloquent\Relations\Relation
+     */
+    public function assignedAccessories()
+    {
+        return $this->morphMany(\App\Models\AccessoryCheckout::class, 'assigned', 'assigned_type', 'assigned_to');
+    }
+
     public function setLdapOuAttribute($ldap_ou)
     {
         return $this->attributes['ldap_ou'] = empty($ldap_ou) ? null : $ldap_ou;
@@ -313,21 +326,4 @@
     {
         return $query->leftJoin('users as location_user', 'locations.manager_id', '=', 'location_user.id')->orderBy('location_user.first_name', $order)->orderBy('location_user.last_name', $order);
     }
-            /**
-     * Establishes the location -> uploads relationship
-     *
-     * @todo I don't think we use this?
-     *
-     * @author A. Gianotto <snipe@snipe.net>
-     * @since [v6.3]
-     * @return \Illuminate\Database\Eloquent\Relations\Relation
-     */
-    public function uploads()
-    {
-        return $this->hasMany(\App\Models\Actionlog::class, 'item_id')
-            ->where('item_type', self::class)
-            ->where('action_type', '=', 'uploaded')
-            ->whereNotNull('filename')
-            ->orderBy('created_at', 'desc');
-    }
 }
